### DISCUSSION TITLE
Allow header variant selection in the API operation details widget

### DISCUSSION THREAD

**Discussion by AnRei123** [Azure/api-management-developer-portal#2557] [Feature request]
#  Bug description

In the API details web widget, the empty space in front of the heading 3 (orange in the example screenshot) is not consistently the same and it is too less to easily understand that a new sub chapter is starting in the following.
![ApiDetailsHeader3emptySpaceBeforeMissing](https://user-images.githubusercontent.com/58981351/96587791-14c0fc00-12e3-11eb-81d4-2003ff5b7f2e.PNG)

DevPortal user complain about the misleading layout within the web widget. I already was forced to change my style configuration of the headings to make it a little bit more obvious which sections do belong together. The complaint came from people who are already familiar with API definitions to a certain extend, but not in all details. And I agree that the layout isn´t optimized to easily and quickly orientate in the API details section.

## Expected behavior

Empty space should be consistently applied before a heading 3 section for all sections of the API details web widget. It should be obvious to the reader of the API details where a new sub chapter starts and where it ends. Sufficient white space in front of each heading level 3 helps the user to better orientate in the document.

## Is your portal managed or self-hosted?

Managed

## Environment

 - Operating system: Windows 10
 - Browser: Chrome (latest version)

**Comment by azaslonov** [comment]
Hi @AnRei123, basically, there are several elements on the page, i.e. "Heading 2" and "Paragraph" - those have their own spacing parameters, and you can control them in the Style guide:
![image](https://user-images.githubusercontent.com/2320302/96771364-64accb00-1396-11eb-8a06-a08c68e53c11.png)


However, we do recognize the problem that, in some cases, individual elements need to have those controls as well, even if they are part of a composite widget like operation details. This problem we're planning to address soon.

**Comment by AnRei123** [comment]
Thanks for your reply. I do already know how to set up the spacing parameters for heading levels. The issue isn´t solved with this answer! If I change the spacing settings centrally, it got reflected all over the place in the portal. But I do NOT want to add further empty space for the other occurrences! And I expect that the layout of the API widget is properly composed based on my chosen spacing. If my spacing does work properly for blocks/sections and text fields, why doesn´t it properly reflects on the API details page? And if I change the spacing of heading 3 in the styles section, still the problem remains that the empty space in front of heading 3 is still not consistently applied for the different sub paragraphs. 
Therefore, I cannot agree that this issue is closed. The bug is still there. We expect a more professional layout when rendering the display for the API details widget based on the centrally configured headings spacings that is also fitting for the other text elements!!!

**Comment by mikebudzynski** [comment]
@AnRei123 From what I understand, allowing to apply a header style only in the API operation details widget would solve your problem. I have reopened this issue and marked it as a feature request.
